src/main/java/com/shaneschulte/mc/clanarena/commands/CreateCmd.java

```diff
@@ -0,0 +1,379 @@
+        }
+
+        // if /ca create <name>
+        else {
+            String name = args[1];
+            currentArenaNameByPlayer.put(p.getName(), args[1]);
+            playersInTheMiddleOfDesignatingThe2PointsNeededToCreateAnArenaRegion.add(p.getName());
+            MsgUtils.success(p,"Creating arena \"" + MsgUtils.Colors.VARIABLE + name + MsgUtils.Colors.SUCCESS + "\"");
+            MsgUtils.sendMessage(p, MsgUtils.Colors.VARIABLE + "Left Click " + MsgUtils.Colors.INFO + "the first point in the " + MsgUtils.Colors.VARIABLE + "Arena" + MsgUtils.Colors.INFO + "'s region");
+            MsgUtils.sendMessage(p, MsgUtils.Colors.VARIABLE + "Right Click " + MsgUtils.Colors.INFO + "the second point in the " + MsgUtils.Colors.VARIABLE + "Arena" + MsgUtils.Colors.INFO + "'s region");
+
+            BossBar bar = Bukkit.getServer().createBossBar("PROGRESS", BarColor.PINK, BarStyle.SEGMENTED_12);
+            currentPlayerBossBars.put(p.getName(), bar);
+            bar.addPlayer(p);
+            updateProgressBossBar(p.getName());
+        }
+
+    }
+
+    /**
+     * Used to set the region in stage 01
+     */
+    @EventHandler
+    public void onPlayerInteract (PlayerInteractEvent event) {
+        // define player
+        Player player = event.getPlayer();
+
+        // if player is in the middle of designating the 2 points
+        if (isPlayerSelectingRegion(player.getName())) {
+
+            // on left click block
+            if (event.getAction() == Action.LEFT_CLICK_BLOCK) {
+                event.setCancelled(true);
+                currentPos1ByPlayer.put(player.getName(), event.getClickedBlock().getLocation());
+                if (isDoneSelectingPositions(player.getName()))
+                    MsgUtils.sendMessage(player, "Type \"" + MsgUtils.Colors.SUCCESS + "done" + MsgUtils.Colors.INFO + "\" in chat when done");
+                updateProgressBossBar(player.getName());
+            }
+
+            // on right click block
+            else if (event.getAction() == Action.RIGHT_CLICK_BLOCK) {
+                event.setCancelled(true);
+                currentPos2ByPlayer.put(player.getName(), event.getClickedBlock().getLocation());
+                if (isDoneSelectingPositions(player.getName()))
+                    MsgUtils.sendMessage(player, "Type \"" + MsgUtils.Colors.SUCCESS + "done" + MsgUtils.Colors.INFO + "\" in chat when done");
+                updateProgressBossBar(player.getName());
+            }
+        }
+    }
+
+    /**
+     * Used to transition from stage 01 -> stage 02 ("done")
+     * Used to transition from stage 02 -> complete ("done")
+     * Used to add spawn points in stage 02 -> "add"
+     * Used to transition which team is having spawn points updated -> "next"
+     */
+    @EventHandler
+    public void onPlayerChat(AsyncPlayerChatEvent event) {
```

Comment:
The complexity of this handler is far too high - you can either have multiple listeners for this event and do one thing in each, or break it out into function calls and leave the high level logic in here. Your call